### PR TITLE
Modify audit and policy

### PR DESCRIPTION
**Pull request by y0urself**
**What**:

Added modifiers for Audits and Policies:
* `modify_audits`
* `modify_policy_set_nvt_preference`
* `modify_policy_set_name`
* `modify_policy_set_comment`
* `modify_policy_set_scanner_preference`
* `modify_policy_set_nvt_selection`
* `modify_policy_set_family_selection`

**Why**:

These API calls are missing.

**How**:

Using the `modify_task` and `modify_config_`

**Checklist**:

- [x] Tests
- [x] [CHANGELOG](https://github.com/greenbone/python-gvm/blob/master/CHANGELOG.md) Entry
- [x] Documentation

### CODE REVIEW

**Review by nichtsfrei — APPROVED** [review]

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 Greenbone Networks GmbH
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:2]:
```suggestion
# Copyright (C) 2020 Greenbone Networks GmbH
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from collections import OrderedDict
+
+from gvm.errors import RequiredArgument, InvalidArgument, InvalidArgumentType
+
+from gvm.protocols.gmpv7 import HostsOrdering
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:25]:
```suggestion
from gvm.protocols.gmpv9 import HostsOrdering
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_comment.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 Greenbone Networks GmbH
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_comment.py:2]:
```suggestion
# Copyright (C) 2020 Greenbone Networks GmbH
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py [hunk 4]
```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 Greenbone Networks GmbH
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py:2]:
```suggestion
# Copyright (C) 2020 Greenbone Networks GmbH
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 Greenbone Networks GmbH
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py:2]:
```suggestion
# Copyright (C) 2020 Greenbone Networks GmbH
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from gvm.errors import RequiredArgument
+
+
+class GmpModifyPolicySetNvtPreferenceTestCase:
+    def test_modify_config_set_nvt_pref(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py:25]:
```suggestion
    def test_modify_policy_set_nvt_pref(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        self.connection.send.has_been_called_with(
+            '<modify_config config_id="c1">'
+            '<preference>'
+            '<nvt oid="o1"/>'
+            '<name>foo</name>'
+            '</preference>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_set_nvt_pref_with_value(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py:50]:
```suggestion
    def test_modify_policy_set_nvt_pref_with_value(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+            '<modify_config config_id="c1">'
+            '<preference>'
+            '<nvt oid="o1"/>'
+            '<name>foo</name>'
+            '<value>YmFy</value>'
+            '</preference>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_set_nvt_pref_missing_nvt_oid(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py:65]:
```suggestion
    def test_modify_policy_set_nvt_pref_missing_nvt_oid(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+            self.gmp.modify_policy_set_nvt_preference(
+                'c1', 'foo', nvt_oid='', value='bar'
+            )
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_nvt_preference(
+                'c1', 'foo', '', value='bar'
+            )
+
+    def test_modify_config_nvt_pref_missing_name(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py:81]:
```suggestion
    def test_modify_policy_nvt_pref_missing_name(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+            self.gmp.modify_policy_set_nvt_preference(
+                'c1', name='', nvt_oid='o1', value='bar'
+            )
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_nvt_preference(
+                'c1', '', nvt_oid='o1', value='bar'
+            )
+
+    def test_modify_config_set_comment_missing_config_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_preference.py:97]:
```suggestion
    def test_modify_policy_set_comment_missing_config_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py [hunk 11]
```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 Greenbone Networks GmbH
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py:2]:
```suggestion
# Copyright (C) 2020 Greenbone Networks GmbH
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from gvm.errors import RequiredArgument, InvalidArgumentType
+
+
+class GmpModifyPolicySetNvtSelectionTestCase:
+    def test_modify_config_set_nvt_selection(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py:25]:
```suggestion
    def test_modify_policy_set_nvt_selection(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+
+        self.connection.send.has_been_called_with(
+            '<modify_config config_id="c1">'
+            '<nvt_selection>'
+            '<family>foo</family>'
+            '</nvt_selection>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_set_nvt_selection_missing_config_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py:88]:
```suggestion
    def test_modify_policy_set_nvt_selection_missing_config_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_nvt_selection(
+                policy_id='', family='foo', nvt_oids=['o1']
+            )
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_nvt_selection('', 'foo', ['o1'])
+
+    def test_modify_config_set_nvt_selection_missing_family(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py:102]:
```suggestion
    def test_modify_policy_set_nvt_selection_missing_family(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_nvt_selection(
+                policy_id='c1', family='', nvt_oids=['o1']
+            )
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_nvt_selection('c1', '', ['o1'])
+
+    def test_modify_config_set_nvt_selection_invalid_nvt_oids(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_nvt_selection.py:116]:
```suggestion
    def test_modify_policy_set_nvt_selection_invalid_nvt_oids(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py [hunk 16]
```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2018 Greenbone Networks GmbH
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py:2]:
```suggestion
# Copyright (C) 2020 Greenbone Networks GmbH
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from gvm.errors import RequiredArgument
+
+
+class GmpModifyPolicySetScannerPreferenceTestCase:
+    def test_modify_config_set_scanner_pref(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py:25]:
```suggestion
    def test_modify_policy_set_scanner_pref(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+
+        self.connection.send.has_been_called_with(
+            '<modify_config config_id="c1">'
+            '<preference>'
+            '<name>foo</name>'
+            '</preference>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_set_scanner_pref_with_value(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py:48]:
```suggestion
    def test_modify_policy_set_scanner_pref_with_value(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py [hunk 19]
```diff
@@ -0,0 +1,90 @@
...
+        self.connection.send.has_been_called_with(
+            '<modify_config config_id="c1">'
+            '<preference>'
+            '<name>foo</name>'
+            '<value>YmFy</value>'
+            '</preference>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_scanner_pref_missing_name(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py:60]:
```suggestion
    def test_modify_policy_scanner_pref_missing_name(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py [hunk 20]
```diff
@@ -0,0 +1,90 @@
...
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_scanner_preference(
+                'c1', name='', value='bar'
+            )
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_scanner_preference('c1', '', value='bar')
+
+    def test_modify_config_set_comment_missing_config_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_scanner_preference.py:74]:
```suggestion
    def test_modify_policy_set_comment_missing_config_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_name.py [hunk 21]
```diff
@@ -0,0 +1,56 @@
...
+    def test_modify_config_set_name(self):
+        self.gmp.modify_policy_set_name('c1', 'foo')
+
+        self.connection.send.has_been_called_with(
+            '<modify_config config_id="c1">'
+            '<name>foo</name>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_set_name_missing_config_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_name.py:34]:
```suggestion
    def test_modify_policy_set_name_missing_config_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_name.py [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_name(policy_id=None, name='name')
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_name('', name='name')
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_name(policy_id='', name='name')
+
+    def test_modify_config_set_name_missing_name(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_name.py:44]:
```suggestion
    def test_modify_policy_set_name_missing_name(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py [hunk 23]
```diff
@@ -0,0 +1,189 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from gvm.errors import RequiredArgument, InvalidArgumentType
+
+
+class GmpModifyPolicySetFamilySelectionTestCase:
+    def test_modify_config_set_family_selection(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py:25]:
```suggestion
    def test_modify_policy_set_family_selection(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py [hunk 24]
```diff
@@ -0,0 +1,189 @@
...
+            '<family>'
+            '<name>bar</name>'
+            '<all>1</all>'
+            '<growing>1</growing>'
+            '</family>'
+            '</family_selection>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_set_family_selection_missing_config_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py:87]:
```suggestion
    def test_modify_policy_set_family_selection_missing_config_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py [hunk 25]
```diff
@@ -0,0 +1,189 @@
...
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_family_selection(
+                policy_id='', families=['foo']
+            )
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_policy_set_family_selection('', ['foo'])
+
+    def test_modify_config_set_family_selection_invalid_families(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py:101]:
```suggestion
    def test_modify_policy_set_family_selection_invalid_families(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py [hunk 26]
```diff
@@ -0,0 +1,189 @@
...
+
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_policy_set_family_selection(
+                policy_id='c1', families=''
+            )
+
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_policy_set_family_selection('c1', '')
+
+    def test_modify_config_set_family_selection_with_auto_add_new_families(
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py:115]:
```suggestion
    def test_modify_policy_set_family_selection_with_auto_add_new_families(
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py [hunk 27]
```diff
@@ -0,0 +1,189 @@
...
+            '<family>'
+            '<name>foo</name>'
+            '<all>1</all>'
+            '<growing>1</growing>'
+            '</family>'
+            '</family_selection>'
+            '</modify_config>'
+        )
+
+    def test_modify_config_set_family_selection_with_auto_add_new_nvts(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_family_selection.py:152]:
```suggestion
    def test_modify_policy_set_family_selection_with_auto_add_new_nvts(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_comment.py [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from gvm.errors import RequiredArgument
+
+
+class GmpModifyPolicySetCommentTestCase:
+    def test_modify_config_set_comment(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_comment.py:25]:
```suggestion
    def test_modify_policy_set_comment(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_policy_set_comment.py [hunk 29]
```diff
@@ -0,0 +1,60 @@
...
+            '</modify_config>'
+        )
+
+        self.gmp.modify_policy_set_comment('c1', comment=None)
+
+        self.connection.send.has_been_called_with(
+            '<modify_config config_id="c1">' '<comment/>' '</modify_config>'
+        )
+
+    def test_modify_config_set_comment_missing_config_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_policy_set_comment.py:48]:
```suggestion
    def test_modify_policy_set_comment_missing_config_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 30]
```diff
@@ -0,0 +1,222 @@
...
+
+
+class GmpModifyAuditCommandTestCase:
+    def test_modify_task(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:29]:
```suggestion
    def test_modify_audit(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 31]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit('t1')
+
+        self.connection.send.has_been_called_with('<modify_task task_id="t1"/>')
+
+    def test_modify_task_missing_task_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:34]:
```suggestion
    def test_modify_audit_missing_task_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 32]
```diff
@@ -0,0 +1,222 @@
...
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_audit(None)
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_audit('')
+
+        with self.assertRaises(RequiredArgument):
+            self.gmp.modify_audit(audit_id='')
+
+    def test_modify_task_with_name(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:44]:
```suggestion
    def test_modify_audit_with_name(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 33]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', name='foo')
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">' '<name>foo</name>' '</modify_task>'
+        )
+
+    def test_modify_task_with_policy_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:51]:
```suggestion
    def test_modify_audit_with_policy_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 34]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', policy_id='c1')
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">' '<config id="c1"/>' '</modify_task>'
+        )
+
+    def test_modify_task_with_target_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:58]:
```suggestion
    def test_modify_audit_with_target_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 35]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', target_id='t1')
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">' '<target id="t1"/>' '</modify_task>'
+        )
+
+    def test_modify_task_with_scanner_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:65]:
```suggestion
    def test_modify_audit_with_scanner_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 36]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', scanner_id='s1')
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">' '<scanner id="s1"/>' '</modify_task>'
+        )
+
+    def test_modify_task_with_schedule_id(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:72]:
```suggestion
    def test_modify_audit_with_schedule_id(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 37]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', schedule_id='s1')
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">' '<schedule id="s1"/>' '</modify_task>'
+        )
+
+    def test_modify_task_with_comment(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:79]:
```suggestion
    def test_modify_audit_with_comment(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 38]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', comment='bar')
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">'
+            '<comment>bar</comment>'
+            '</modify_task>'
+        )
+
+    def test_modify_task_with_alerts_ids(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:88]:
```suggestion
    def test_modify_audit_with_alerts_ids(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 39]
```diff
@@ -0,0 +1,222 @@
...
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">'
+            '<alert id="a1"/>'
+            '<alert id="a2"/>'
+            '<alert id="a3"/>'
+            '</modify_task>'
+        )
+
+    def test_modify_task_invalid_alerts_ids(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:99]:
```suggestion
    def test_modify_audit_invalid_alerts_ids(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 40]
```diff
@@ -0,0 +1,222 @@
...
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_audit(audit_id='t1', alert_ids='')
+
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_audit(audit_id='t1', alert_ids='a1')
+
+    def test_modify_task_with_empty_alert_ids(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:106]:
```suggestion
    def test_modify_audit_with_empty_alert_ids(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 41]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', alert_ids=[])
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">' '<alert id="0"/>' '</modify_task>'
+        )
+
+    def test_modify_task_with_alterable(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:113]:
```suggestion
    def test_modify_audit_with_alterable(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 42]
```diff
@@ -0,0 +1,222 @@
...
+
+        self.gmp.modify_audit(audit_id='t1', alterable=False)
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">'
+            '<alterable>0</alterable>'
+            '</modify_task>'
+        )
+
+    def test_modify_task_with_hosts_ordering(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:130]:
```suggestion
    def test_modify_audit_with_hosts_ordering(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 43]
```diff
@@ -0,0 +1,222 @@
...
+            audit_id='t1', hosts_ordering=HostsOrdering.REVERSE
+        )
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">'
+            '<hosts_ordering>reverse</hosts_ordering>'
+            '</modify_task>'
+        )
+
+    def test_modify_task_invalid_hosts_ordering(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:141]:
```suggestion
    def test_modify_audit_invalid_hosts_ordering(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 44]
```diff
@@ -0,0 +1,222 @@
...
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_audit(audit_id='t1', hosts_ordering='foo')
+
+    def test_modify_task_with_schedule(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:145]:
```suggestion
    def test_modify_audit_with_schedule(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 45]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', schedule_id='s1')
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">' '<schedule id="s1"/>' '</modify_task>'
+        )
+
+    def test_modify_task_with_schedule_periods(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:152]:
```suggestion
    def test_modify_audit_with_schedule_periods(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 46]
```diff
@@ -0,0 +1,222 @@
...
+
+        self.gmp.modify_audit(audit_id='t1', schedule_periods=5)
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">'
+            '<schedule_periods>5</schedule_periods>'
+            '</modify_task>'
+        )
+
+    def test_modify_task_invalid_schedule_periods(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:169]:
```suggestion
    def test_modify_audit_invalid_schedule_periods(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 47]
```diff
@@ -0,0 +1,222 @@
...
+        with self.assertRaises(InvalidArgument):
+            self.gmp.modify_audit(audit_id='t1', schedule_periods='foo')
+
+        with self.assertRaises(InvalidArgument):
+            self.gmp.modify_audit(audit_id='t1', schedule_periods=-1)
+
+    def test_modify_task_with_observers(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:176]:
```suggestion
    def test_modify_audit_with_observers(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 48]
```diff
@@ -0,0 +1,222 @@
...
+        self.gmp.modify_audit(audit_id='t1', observers=['u1', 'u2'])
+
+        self.connection.send.has_been_called_with(
+            '<modify_task task_id="t1">'
+            '<observers>u1,u2</observers>'
+            '</modify_task>'
+        )
+
+    def test_modify_task_invalid_observers(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:185]:
```suggestion
    def test_modify_audit_invalid_observers(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 49]
```diff
@@ -0,0 +1,222 @@
...
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_audit(audit_id='t1', observers='')
+
+        with self.assertRaises(InvalidArgumentType):
+            self.gmp.modify_audit(audit_id='t1', observers='foo')
+
+    def test_modify_task_with_preferences(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:192]:
```suggestion
    def test_modify_audit_with_preferences(self):
```

File: tests/protocols/gmpv9/testcmds/test_modify_audit.py [hunk 50]
```diff
@@ -0,0 +1,222 @@
...
+            '</preference>'
+            '<preference>'
+            '<scanner_name>lorem</scanner_name>'
+            '<value>ipsum</value>'
+            '</preference>'
+            '</preferences>'
+            '</modify_task>'
+        )
+
+    def test_modify_task_invalid_preferences(self):
```
Comment (bjoernricks) [inline tests/protocols/gmpv9/testcmds/test_modify_audit.py:213]:
```suggestion
    def test_modify_audit_invalid_preferences(self):
```

**Review by bjoernricks — CHANGES_REQUESTED** [review]
Sorry i wasn't fast enough. The PR contains a lot of copy&paste mistakes.

**Pull request merged**